tests/charms/tls_certificates_interface/v0/test_tls_certificates.py

```diff
@@ -318,3 +320,17 @@ def test_given_valid_relation_data_when_on_relation_changed_then_certificate_ava
             )
         ]
         patch_emit.assert_has_calls(calls, any_order=True)
+
+        self.charm.framework.model.relations = {"certificates": [event.relation]}
```

Comment:
This should be a test of its own, let's not bundle multiple behaviour validation in a single test